### PR TITLE
DC-244: Enhance offheap queue

### PR DESCRIPTION
**Pull request by freddy61025**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

- https://issues.opennms.org/browse/DC-244
- This PR is only adding a new queue in OpenNMS without actual implementation. The real usage will be in https://issues.opennms.org/browse/DC-283.

### CODE REVIEW

**Comment by patrick-schweizer** [comment]
I wonder if the PR shouldn't go against release-30.x branch? To my knowledge this is what we are aiming. At least the cloud plugin develops against it.

**Comment by freddy61025** [comment]
> I wonder if the PR shouldn't go against release-30.x branch? To my knowledge this is what we are aiming for. At least the cloud plugin develops against it.

Let me confirm with Maxim again. Last week we still waiting for the production to confirm. If it need to be in release-30.x we can cherry-pick.

**Review by patrick-schweizer — DISMISSED** [review]
I don't claim I fully understand what the code does but it looks good to me from what I can see.
It might be nice to have somewhere a summary of the concept.

File: core/ipc/sink/off-heap/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DataBlocksOffHeapQueue logic example
```
Comment (patrick-schweizer) [inline core/ipc/sink/off-heap/README.md:1]: nice! :-)

**Pull request merged**